### PR TITLE
Social Share Tray component

### PR DESCRIPTION
**Pull request by mendelB**
### What does this PR do?

This PR adds a new `SocialShareTray` component!

### Any background context you want to provide?
This component is based on part of the design for the refurbished `SocialDriveAction`, which calls for a new 'tray' of social share buttons.

(It's still missing the click handlers for the Facebook Messenger button, working through some issues there.)

Here's how it'd look inside the Social Drive Action:

**LARGE**
image

**MEDIUM**
image

**SMALL**
image

Responsiveness GIF

~As you can see in the gif, and the medium pic, sometimes one of the buttons gets booted to the next row so that you have 3 on the first row, and 1 on the second.~
~@weerd mentioned to me this is not ideal, but I, unfortunately, wasn't able to conjure up a graceful way to address this yet 😖. @lkpttn mentioned eventually changing up some of our margin settings so that the content itself doesn't get too constrained, but rather the margin adjusts itself appropriately.~
UPDATE: Hiding text on mobile *and* tablet. Added a media query fix for desktop.

### What are the relevant tickets/cards?

Refs [Pivotal ID #159445991](https://www.pivotaltracker.com/story/show/159445991)

### Checklist

* [ ] Documentation added for new features/changed endpoints.
* [ ] Added appropriate feature/unit tests.
* [x] Added screenshot to PR description of related front-end updates on **small** screens.
* [x] Added screenshot to PR description of related front-end updates on **medium** screens.
* [x] Added screenshot to PR description of related front-end updates on **large** screens.

### CODE REVIEW

File: resources/assets/components/utilities/SocialShareTray/social-share-tray.scss [hunk 1]
```diff
@@ -0,0 +1,80 @@
+@import '../../../scss/next-toolbox.scss';
+
+// Sets a background color for social share buttons without overriding
+// the inherited 'disabled' and 'hover' .button states
+@mixin button-background($color) {
```
Comment (mendelB) [inline resources/assets/components/utilities/SocialShareTray/social-share-tray.scss]: I was a bit adventurous here, but looking for a way to eliminate calling this for each social button. Hope this was appropriate, slash - should this go in `next-toolbox.scss` or something?

File: resources/assets/components/utilities/SocialShareTray/social-share-tray.scss [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  &:not([disabled]) {
+    background-color: $color;
+
+    &:hover {
+      background-color: lighten($color, $color-tint);
+    }
+  }
+}
+
+// Social share button background colors
```
Comment (mendelB) [inline resources/assets/components/utilities/SocialShareTray/social-share-tray.scss]: Should these go in `next-toolbox.scss`? I wasn't sure if these were just specific to this component

**Comment by weerd** [comment]
> (It's still missing the click handlers for the Facebook Messenger button, working through some issues there.)

Are we going to refrain from merging until this is resolved?

**Comment by weerd** [comment]
> @weerd mentioned to me this is not ideal, but I, unfortunately, wasn't able to conjure up a graceful way to address this yet 😖. @lkpttn mentioned eventually changing up some of our margin settings so that the content itself doesn't get too constrained, but rather the margin adjusts itself appropriately.

As a "for now" fix, can we just collapse the buttons earlier and remove the text at both medium AND small? Only show the text at large sizes?

File: resources/assets/components/utilities/SocialShareTray/social-share-tray.scss (hunk 2 above)
Comment (weerd) [reply resources/assets/components/utilities/SocialShareTray/social-share-tray.scss]: Yeah, might be best in the tool box!

File: resources/assets/components/utilities/SocialShareTray/social-share-tray.scss (hunk 1 above)
Comment (weerd) [reply resources/assets/components/utilities/SocialShareTray/social-share-tray.scss]: I think this works. We could try something in the future with really utilizing the `currentColor` keyword in CSS to inherit from the parent colors or whatever, but I've only seen tutorials on it and never implemented it myself; but seeeeeeems really useful! I [watched](https://www.youtube.com/watch?v=krbKkLPXwlQ) it a while ago 😬

File: resources/assets/components/utilities/SocialShareTray/social-share-tray.scss [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        @include button-background($email-background-color);
+      }
+
+      height: 55px;
+      padding-right: 20px;
+      padding-left: 20px;
+      text-transform: none;
+      font-size: $font-regular;
+
+      img.icon-image {
```
Comment (weerd) [inline resources/assets/components/utilities/SocialShareTray/social-share-tray.scss]: don't use the tag + class name together `img.icon-image`. Fine to just use `.icon-image`. Over-specifying means it's much hard to override if needed to later!

File: resources/assets/components/utilities/SocialShareTray/social-share-tray.scss [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+// Social share button background colors
+$email-background-color: #0ab6fe;
+$twitter-background-color: #00aced;
+$facebook-background-color: #39579a;
+$messenger-background-color: #0084ff;
+
+.social-share-tray {
+  .share-buttons {
+    button.share-button {
```
Comment (weerd) [inline resources/assets/components/utilities/SocialShareTray/social-share-tray.scss]: This [comment](https://github.com/DoSomething/phoenix-next/pull/1072/files#r212423193) applies here too!

File: resources/assets/components/utilities/SocialShareTray/social-share-tray.scss (hunk 1 above)
Comment (mendelB) [reply resources/assets/components/utilities/SocialShareTray/social-share-tray.scss]: That seems really interesting! I made a note to check out that video and follow up!

**Comment by mendelB** [comment]
@weerd I do have the messenger click handler ready, should I just wait with that for a separate PR though? It's gonna be slightly contentious 😬

**Comment by weerd** [comment]
Separate PR ya.

**Pull request merged**